google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -1217,6 +1227,13 @@ def to_gapi
             @gapi
           end
 
+          def time_partitioning type: "DAY", expiration_ms: nil
```

Comment:
Hmmm, although we generally present seconds, it looks like [`Dataset#default_expiration`](https://github.com/GoogleCloudPlatform/google-cloud-ruby/blob/master/google-cloud-bigquery/lib/google/cloud/bigquery/dataset.rb#L149-L172) preserves the milliseconds of the underlying API. I suppose we will have to debate this in your PR.